src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -72,19 +71,35 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
             root = root.TrackNodes(memberNodes);
             var sourceDoc = _document.WithSyntaxRoot(root);
 
+            if (!moveOptions.IsNewType)
+            {
+                // we already have our destination type
+                var fixedSolution = await RefactorAndMoveAsync(
+                    moveOptions.SelectedMembers,
+                    memberNodes,
+                    sourceDoc.Project.Solution,
+                    moveOptions.Destination!,
+                    // TODO: Find a way to merge/change generic type args for classes, or change PullMembersUp to handle instead
+                    typeArgIndices: ImmutableArray<int>.Empty,
```

Comment:
I don't have a good solution for this scenario:
1. User has a source type with generic type parameters: `Class1<T>`
2. Selected member uses the type parameters in a way that they must be explicit/can't be implied: `public static T TestMethod()`
3. Destination has no type parameters or a different set of type params from the source type: `Class2`

How do we refactor usages of this method? How do we qualify the type parameter when we move to the destination?

We likely would need some way to merge or condense type parameters.

This should be a fairly rare case, as it's unlikely a static member of a type uses class-level type parameters. Generally a user would probably put them on the method itself. However, a good example is something like `ImmutableArray<T>.Create()`

Currently, this implementation removes all type params when refactoring, and does not qualify any type params when moving to an existing type.

Reply:
#56174 Is related, but I think there's two separate problems here, as here we need to requalify references to the moved members using type arguments (if they're not implied), including all the existing destination type arguments that were already being used.